src/plugins/visualizations/public/visualize_app/utils/get_top_nav_config.tsx

```diff
@@ -272,6 +282,35 @@ export const getTopNavConfig = (
     visualizeCapabilities.save || (!originatingApp && dashboardCapabilities.showWriteControls);
 
   const topNavMenu: TopNavMenuData[] = [
+    ...(displayEditInLensItem
+      ? [
+          {
+            id: 'goToLens',
+            label: i18n.translate('visualizations.topNavMenu.goToLensButtonLabel', {
+              defaultMessage: 'Edit visualization in Lens',
+            }),
+            emphasize: false,
+            description: i18n.translate('visualizations.topNavMenu.goToLensButtonAriaLabel', {
+              defaultMessage: 'Go to Lens with your current configuration',
+            }),
+            disableButton: !editInLensConfig,
+            testId: 'visualizeEditInLensButton',
+            run: async () => {
+              const updatedWithMeta = {
+                ...editInLensConfig,
+                savedObjectId: visInstance.vis.id,
+                embeddableId,
+                vizEditorOriginatingAppUrl: getVizEditorOriginatingAppUrl(history),
+                originatingApp,
+              };
+              if (editInLensConfig) {
+                hideLensBadge();
+                getUiActions().getTrigger(VISUALIZE_EDITOR_TRIGGER).exec(updatedWithMeta);
+              }
+            },
+          },
```

Comment:
Similar to my previous comment, would it be possible to custom add a border separating this button from the other actions in the white header in the top-right of the interface (to match the designs)?